### DISCUSSION TITLE
Generate Docker image using updated source

### DISCUSSION THREAD

**Discussion by NuwanSameera** [rabbitmq/rabbitmq-server#5251] [Other] [answered]
I did some modifications in rabbitmq_management plugin for fulfill my use case. All the codes are working with `make run-broker`. Now I need to build docker image using modified code. Please provide me documentation or steps need to follow.

**Comment by lukebakken** [comment, marked as answer]
My guess is the easiest way will be to use the official Docker image as your base image, then overwrite the `rabbitmq_management` plugin in that image with your code:

```
cd path/to/rabbitmq/rabbitmq-server
git checkout your-branch-with-changes
cd deps/rabbitmq_management
make dist
```

Copy the contents of `plugins/rabbitmq_management-*` into your Docker image via a `COPY` statement in the `Dockerfile`.

**Reply by NuwanSameera** [reply]
Thank you. I did it. But have a problem. 

I added new URL to management plugin to delete connection by user name. I tested it with make run-broker. All are working properly. Then I build plugins and replace existing plugins folder with new one. Management console loading but I cannot access new API. It is DELETE method. After calling it given 405 Method Not Allowed. .beam files also available in ebin folder. What is the reason for this ?

**Reply by lukebakken** [reply]
> Management console loading but I cannot access new API. It is DELETE method. After calling it given 405 Method Not Allowed

If you'd like to share your code and docker file, I could probably debug this. Otherwise, I'm just guessing at the cause.

My first guess is that since it works via `make run-broker` but not in your Docker image, that the original `.beam` file is still being used.

**Reply by NuwanSameera** [reply]
Sure. This is my code

https://github.com/NuwanSameera/rabbitmq-server/tree/feature-delete-connection-by-name

This is my Dockerfile

```
FROM rabbitmq:3.10.6

RUN rm -r plugins

RUN mkdir plugins

ADD plugins  plugins

```

**Reply by lukebakken** [reply]
Check this out:

https://github.com/lukebakken/rabbitmq-server-5251/blob/main/Dockerfile

I didn't test it because your API call is a bit confusing. If you want to delete a connection by user name, I would expect a user name as the parameter to the API call, not the connection name.

Are you really deleting all connections for a user based on the name of one of those connections?

At any rate, try out the code I provided and let me know what you think.

**Reply by NuwanSameera** [reply]
Thank you for your answer and suggestions. 

Problem is I copied plugins to wrong folder `./plugins`. It should copy to `/opt/rabbitmq/plugins`
**If you want to delete a connection by user name** -> I will work on this

**Are you really deleting all connections for a user based on the name of one of those connections?**  -> Yes. I have specific use case like this. There are telemetry devices and they connected through SSL enabled MQTT connections. We enabled cert-login feature and there is custom authentication logic based on CN key of the certificate. So in our case there is only one MQTT connection for given user name ie: CN key. If some reason we need to force close existing connection only CN key is the known parameter.  In this case we need to force close all existing connections for given user name.
